### ISSUE TITLE
Terminal won't exit after calling xclip

### ISSUE THREAD

**Issue by SamHH** [kovidgoyal/kitty#828]
Run: `echo example | xclip -selection c`

The terminal will not exit either with `exit` or Ctrl+D. To close the terminal, it must either be forcefully killed, or you must open another terminal (any, not necessarily Kitty) and re-run xclip.

I do not experience this behaviour in either Alacritty or Termite.

**Comment by maximbaz** [comment]
`xclip` is annoying in this regard, it blocks stdout. Your options are:

- use `-f` flag for `xclip`
- pipe xclip's stdout to /dev/null
- use kitty's option [close_on_child_death](https://sw.kovidgoyal.net/kitty/conf.html#opt-kitty.close_on_child_death)
- use `xsel`

**Comment by SamHH** [comment]
xclip's `-f` flag doesn't fix this issue for me, it just prints out before exhibiting the same behaviour.

I wonder why Kitty has this issue but not the other terminals I mentioned? What are the effects of enabling `close_on_child_death`?

**Comment by maximbaz** [comment]
Click on the link, there is a description 😉

**Comment by SamHH** [comment]
Haha yes, I'm not super familiar with the terminology and wondering in layman's terms why that's not set as the default :stuck_out_tongue:

**Comment by kovidgoyal** [comment]
Yes, terminals that close even when some process is holding onto to stdout/stderr mean that there is now now way to see output from those processes,and those processes might crash because stdout/stderr no longer work for them. This is a bad design in most terminal emulators that encourages the sue of sloppy programming practices such as used in xclip.

Incidentally you dont need to use xclip with kitty. Use the clipboard kitten instead, which does not suffer from these design flaws.

**Comment by SolitudeSF** [comment]
advantage of xclip over clipboard kitten is that its much faster. it finishes in under 5 ms, while kitten runs over 160ms, which adds considerable lag when invoked from text editor etc.

**Comment by kovidgoyal** [comment]
That's just the time it takes to load python and the kitty module. If it bothers you, you can write a shell script or a pure python script to copy to the clipboard in a few dozen lines that will not have the startup penalty.

**Comment by kovidgoyal** [comment]
Oh and if you are using python 3.7 and run kitty from master, kitten startup time should be reduced by 30-40%

**Comment by timvisee** [comment]
I would like to add that this seems to happen in bash subshells as well, meaning that `echo $(echo test | xclip -sel clip)` gets stuck as well without the `-f` flag. Not running in a subshell works fine, and `fish` shell doesn't seem to have this issue. 

Binaries invoking `xclip` as subprocess are affected as well. 

I had this issue in `ffsend`:  
https://gitlab.com/timvisee/ffsend/issues/93

**Comment by sathishmanohar** [comment]
The `-f` flag didn't work for me. Piping to `/dev/null` did the trick.

My working piece of code.
`xclip -r -sel clip 1> /dev/null 2> /dev/null`
